File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // input
    string text = get_string("Text: ");

    // declaring the variables we will use in the for loop
    int letters = 0;
    int words = 1;
    int sentences = 0;

    //iterate over teh length of text
    for (int i = 0; i < strlen(text); i++)
    {

        //check if the charachter is alphabetical if so increment letter
        if (isalpha(text[i]))
        {
            letters++;
        }

        //otherwise if the charachter is white space increment word
        else if (text[i] == ' ')
        {
            words++;
        }

        // lastly otherwise if text is . ! or ? increment sentence
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    //here are the math functions you are asked to do
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    //declaring an index of the above
    int index = round(0.0588 * L - 0.296 * S - 15.8);


    //checking if the index is less than one then output that text and if index is greater than 16 then output that and so on
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }



}

File: sentimental-readability/readability.py

# ask user for input
text = input("Text: ")


# declare letters, words and sentences we will increment inside the loop
letters = 0.00
words = 1.00
sentences = 0.00

# loop in range of the length of text
for i in range(len(text)):
    # if text is alphabetical increment letters
    if text[i].isalpha():
        letters += 1
    # else if text is equal to white space increment words
    elif text[i] == ' ':
        words += 1
    # else if therese a dot, exclamation mark, or ! increment sentences.
    elif text[i] == '.' or text[i] == '!' or text[i] == '?':
        sentences += 1

# math function for calculating the letters and sentences
l = letters / words * 100
s = sentences / words * 100

# index of the above
index = round(0.0588 * l - 0.296 * s - 15.8)

# check if index is less than one that means your text is before grade 1
if index < 1:
    print("Before Grade 1")
# else if index is greater than 16 then grade 16+
elif index > 16:
    print("Grade 16+")
# else print the grade and index
else:
    print("Grade ", index)
